sklearn/pipeline.py

```diff
@@ -831,3 +841,114 @@ def make_union(*transformers, **kwargs):
         raise TypeError('Unknown keyword arguments: "{}"'
                         .format(list(kwargs.keys())[0]))
     return FeatureUnion(_name_estimators(transformers), n_jobs=n_jobs)
+
+
+class ColumnTransformer(FeatureUnion):
+    """Applies transformers to columns of a array / dataframe / dict.
+
+    This estimator applies transformer objects to columns or fields of the
+    input, then concatenates the results. This is useful for heterogeneous or
+    columnar data, to combine several feature extraction mechanisms into a
+    single transformer.
+
+    Read more in the :ref:`User Guide <column_transformer>`.
+
+    Parameters
+    ----------
+    transformers : list of tuples
+        List of (name, transformer, column) tuples specifying the transformer
+        objects to be applied to subsets of the data. The columns can be
+        specified as a scalar or list (for multiple columns) of integer or
+        string values. Integers are interpreted as the positional columns,
+        strings as the keys of `X`. In case of positional integers, also
```

Comment:
what does keys mean? should be clear what datatypes we support for X, e.g. mappings, dataframes, struct arrays? Particularly need to be clear if we do not override `fit`, `fit_transform`, `transform` and their docstrings.